src/main/java/com/duke/bot/Duke.java

```diff
@@ -20,144 +47,110 @@ public Duke() {
-                ui.setToken("\\p{javaWhitespace}+");
-                storage.saveToFile(tasks.printList());
-                echo();
-                break;
-
-            case "event":
-                ui.setToken("\\s*/at\\s*|\\n");
-                action = ui.getNext();
-                dateInput = ui.getNext();
-                LocalDate atDate = LocalDate.parse(dateInput);
-                EventTask eventTask = EventTask.createEventTask(action, atDate);
-                tasks.addTask(eventTask);
-                ui.print("Added: " + eventTask.toString());
-                ui.print(String.format("Now you have %d task(s) on your list.\n", tasks.getSize()));
-                ui.printEmptyLine();
-                ui.setToken("\\p{javaWhitespace}+");
-                storage.saveToFile(tasks.printList());
-                echo();
-                break;
-
-            case "find":
-                ui.resetScanner();
-                String keyword = ui.getNextLine();
-                for (int i = 0; i < tasks.getSize(); ++i) {
-                    Task task = tasks.getTask(i);
-                    if (task.getTaskName().contains(keyword)) {
-                        ui.print((i + 1) + ". " + task.toString());
-                    }
-                }
-                echo();
-                break;
-
-            case "bye":
-                ui.printByeMsg();
-                break;
-
-            case "delete":
-                int delIdx = ui.getNextInt() - 1;
-                if (delIdx >= tasks.getSize() || delIdx < 0) {
-                    throw new DukeException("Oops! Target object is out of bounds!");
-                }
-                Task delTask = tasks.getTask(delIdx);
-                tasks.deleteTask(delIdx);
-                ui.print(String.format("Deleted: %s\n\n", delTask.toString()));
-                storage.saveToFile(tasks.printList());
-                echo();
-                break;
-
-            default:
-                throw new DukeException(
-                        "Oops! Invalid commmand word, perhaps you would want to try on of the following: "
-                                + "1. todo 2.deadline 3.event 4.list 5.done 6.bye");
-
-            }
-
-        } catch (DukeException ex) {
-            ui.printError(ex.getMessage());
-            ui.printEmptyLine();
+        String output = Parser.parse(userCommand, ui, storage, tasks);
+        ui.print(output);
+        if (!userCommand.equals("bye")) {
             echo();
         }
     }
 
-
     /**
      * The main method to simulate and run Duke Bot.
      *
      * @param args The arguments written by the user in the command line.
      */
     public static void main(String[] args) {
 
-        Duke duke  = new Duke();
+        Duke duke  = new Duke(new DukeUi());
 
         duke.ui.greet();
 
-        duke.echo();
+        try {
+            duke.echo();
+        } catch (DukeException e) {
+            duke.ui.print(e.getMessage());
```

Comment:
Maybe you can think about how to handle the error message in a better way to make your application work more smoothly?